File: Objective_C_Project/MVC/Classes/Common/Tools/DLNetworkTools.m
//
//  DLNetworkTools.m
//  DLNetEaseNews
//
//  Created by NowOrNever on 11/03/2017.
//  Copyright © 2017 NowOrNever. All rights reserved.
//

#import "DLNetworkTools.h"

@implementation DLNetworkTools

+ (instancetype)sharedNetWorkTools{
    static DLNetworkTools *instance;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        instance = [[self alloc] initWithBaseURL:[NSURL URLWithString:@"http://c.m.163.com/nc/article/list/"]];
    });
    return instance;
}

- (void)requestWithType:(DLRequestType)requestType andUrlString:(NSString *)urlString andParams:(id)parameters andSuccess:(void (^)(id))successBlock andFailture:(void (^)(NSError *))failtureBlock{
    if (requestType == DLRequestTypeGET) {
        [self GET:urlString parameters:parameters progress:nil success:^(NSURLSessionDataTask * _Nonnull task, id  _Nullable responseObject) {
            successBlock(responseObject);
        } failure:^(NSURLSessionDataTask * _Nullable task, NSError * _Nonnull error) {
            failtureBlock(error);
        }];
    }else if (requestType == DLRequestTypePOST){
        [self POST:urlString parameters:parameters progress:nil success:^(NSURLSessionDataTask * _Nonnull task, id  _Nullable responseObject) {
            successBlock(responseObject);
        } failure:^(NSURLSessionDataTask * _Nullable task, NSError * _Nonnull error) {
            failtureBlock(error);
        }];
    }
    
//    [AFHTTPSessionManager manager] GET:<#(nonnull NSString *)#> parameters:<#(nullable id)#> progress:<#^(NSProgress * _Nonnull downloadProgress)downloadProgress#> success:<#^(NSURLSessionDataTask * _Nonnull task, id  _Nullable responseObject)success#> failure:<#^(NSURLSessionDataTask * _Nullable task, NSError * _Nonnull error)failure#>
}

@end

File: SwiftProject/MVVM/Classes/Common/Tools/DLNetworkTools.swift
//
//  DLNetworkTools.swift
//  Swift网络请求封装
//
//  Created by NowOrNever on 27/03/2017.
//  Copyright © 2017 Focus. All rights reserved.
//

import UIKit
import AFNetworking

//在swift里面，枚举可以不需要指定类型
//enum RequestType: Int{

enum RequestType {
    //枚举名需要小写
    case get
    case post
}

class DLNetworkTools: AFHTTPSessionManager {
    static let sharedTools: DLNetworkTools = {
        let tools = DLNetworkTools()
        tools.responseSerializer.acceptableContentTypes?.insert("text/html")
        tools.responseSerializer.acceptableContentTypes?.insert("text/plain")
        return tools
    }()
    
    fileprivate func request(type:RequestType,urlString:String,params:Any?,finishCallBack:@escaping (Any?,Error?)->()){
        if type == .post {
            post(urlString, parameters: params, progress: nil, success: { (_, response) -> () in
                finishCallBack(response,nil);
            }, failure: { (_, error) in
                finishCallBack(nil, error);
            })
        }else{
            get(urlString, parameters: params, progress: nil, success: { (_, response) in
                finishCallBack(response,nil);
            }, failure: { (_, error) in
                finishCallBack(nil, error);
            })
        }
    }
}








